src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/permute/permute_kernel_ref.cpp

```diff
@@ -33,14 +34,44 @@ CommonDispatchData PermuteKernelRef::SetDefault(const permute_params& params) co
     CommonDispatchData dispatchData;
     auto in_layout = params.inputs[0].GetLayout();
     auto out_layout = params.outputs[0].GetLayout();
-    std::vector<std::vector<Tensor::DataChannelName>> dims_by_gws = {{Tensor::DataChannelName::X},
-                                                                     {Tensor::DataChannelName::Y, Tensor::DataChannelName::Z, Tensor::DataChannelName::W},
-                                                                     {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
 
-    const auto& in =  params.inputs[0];
+    std::function<bool(const std::vector<uint16_t>&)> f_to_x = [](const std::vector<uint16_t>& order) {
+        if ((int32_t) order[1] == order.size() - 1) return true;
+        return true;
```

Comment:
I see return true for both return statements. Maybe your intension was 'return false;' for one of the two return statements?